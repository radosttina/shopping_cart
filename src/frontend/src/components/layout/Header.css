.logout_btn {
    border: none;
    border-radius: 0.25rem;
    color: darkcyan;
}

.nav-bar-position {
    position: fixed;
    width: 100%
}