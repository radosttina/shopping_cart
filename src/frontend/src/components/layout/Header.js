import React, { Component } from "react";
import { Link } from "react-router-dom";
import { connect, ReactReduxContext } from "react-redux";
import { logoutUser } from "../../actions/users.js";
import PropTypes from "prop-types";
import "./Header.css";

class Header extends Component {
  render() {
    const guestLinks = (
      <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
        <li className="nav-item">
          <Link to="/register" className="nav-link">
            Register
          </Link>
        </li>
        <li className="nav-item">
          <Link to="/login" className="nav-link">
            Login
          </Link>
        </li>
      </ul>
    );

    const userLinks = (
      <React.Fragment>
        <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
          {this.props.isAdmin ? (
            <li className="nav-item">
              <Link to="/add-product" className="nav-link">
                Add Product
              </Link>
            </li>
          ) : (
            ""
          )}

          <li className="nav-item">
            <Link to="/cart" className="nav-link glyphicon glyphicon-shopping-cart">
              <span className="glyphicon glyphicon-shopping-cart">Cart</span>
            </Link>
          </li>
        </ul>
        <span>{this.props.username ? `Hi, ${this.props.username}  ` : ""}</span>
        <button className="nav-link logout_btn" onClick={this.props.logoutUser}>
          Logout
        </button>
      </React.Fragment>
    );

    return (
      <nav className="navbar navbar-expand-lg navbar-light bg-light justify-content-between nav-bar-position">
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarTogglerDemo03"
          aria-controls="navbarTogglerDemo03"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon"></span>
        </button>
        <Link to="/" className="navbar-brand">
          Shoppify
        </Link>

        <div className="collapse navbar-collapse" id="navbarTogglerDemo03">
          {this.props.isAuthenticated ? userLinks : guestLinks}
        </div>
      </nav>
    );
  }
}

Header.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool,
  isAdmin: PropTypes.bool,
  username: PropTypes.string
}

const mapStateToProps = (state) => ({
  isAuthenticated: state.user.isAuthenticated,
  isAdmin: state.user.is_superuser,
  username: state.user.username,
});

export default connect(mapStateToProps, { logoutUser })(Header);
