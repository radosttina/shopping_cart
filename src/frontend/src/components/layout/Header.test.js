import React from "react";
import Header from "./Header";
import renderer from "react-test-renderer";
import { render } from "../../../test-utils";
import { HashRouter as Router, Route } from "react-router-dom";

test("Header should render links for unauthenticated users.", () => {
  const initialState = {
    user: {
      isAuthenticated: false,
      isAdmin: false,
      username: null,
    },
  };

  const component = render(
    <Router>
      <Header />
    </Router>,
    { initialState: initialState }
  );

  let tree = component.container;
  expect(tree).toMatchSnapshot();
});

test("Header should render links for authenticated non-admin users.", () => {
  const initialState = {
    user: {
      isAuthenticated: true,
      isAdmin: false,
      username: "User",
    },
  };

  const component = render(
    <Router>
      <Header />
    </Router>,
    { initialState: initialState }
  );

  let tree = component.container;
  expect(tree).toMatchSnapshot();
});

test("Header should render links for authenticated admin users.", () => {
  const initialState = {
    user: {
      isAuthenticated: true,
      isAdmin: false,
      username: "Admin",
    },
  };

  const component = render(
    <Router>
      <Header />
    </Router>,
    { initialState: initialState }
  );

  let tree = component.container;
  expect(tree).toMatchSnapshot();
});

