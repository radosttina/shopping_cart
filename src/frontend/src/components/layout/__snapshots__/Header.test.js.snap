// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render links for authenticated admin users. 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light justify-content-between nav-bar-position"
  >
    <button
      aria-controls="navbarTogglerDemo03"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarTogglerDemo03"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <a
      class="navbar-brand"
      href="#/"
    >
      Shoppify
    </a>
    <div
      class="collapse navbar-collapse"
      id="navbarTogglerDemo03"
    >
      <ul
        class="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        
        <li
          class="nav-item"
        >
          <a
            class="nav-link glyphicon glyphicon-shopping-cart"
            href="#/cart"
          >
            <span
              class="glyphicon glyphicon-shopping-cart"
            >
              Cart
            </span>
          </a>
        </li>
      </ul>
      <span>
        Hi, Admin  
      </span>
      <button
        class="nav-link logout_btn"
      >
        Logout
      </button>
    </div>
  </nav>
</div>
`;

exports[`Header should render links for authenticated non-admin users. 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light justify-content-between nav-bar-position"
  >
    <button
      aria-controls="navbarTogglerDemo03"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarTogglerDemo03"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <a
      class="navbar-brand"
      href="#/"
    >
      Shoppify
    </a>
    <div
      class="collapse navbar-collapse"
      id="navbarTogglerDemo03"
    >
      <ul
        class="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        
        <li
          class="nav-item"
        >
          <a
            class="nav-link glyphicon glyphicon-shopping-cart"
            href="#/cart"
          >
            <span
              class="glyphicon glyphicon-shopping-cart"
            >
              Cart
            </span>
          </a>
        </li>
      </ul>
      <span>
        Hi, User  
      </span>
      <button
        class="nav-link logout_btn"
      >
        Logout
      </button>
    </div>
  </nav>
</div>
`;

exports[`Header should render links for unauthenticated users. 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-light bg-light justify-content-between nav-bar-position"
  >
    <button
      aria-controls="navbarTogglerDemo03"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarTogglerDemo03"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <a
      class="navbar-brand"
      href="#/"
    >
      Shoppify
    </a>
    <div
      class="collapse navbar-collapse"
      id="navbarTogglerDemo03"
    >
      <ul
        class="navbar-nav mr-auto mt-2 mt-lg-0"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/register"
          >
            Register
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/login"
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
